src/iadmin.cpp

```diff
@@ -147,6 +149,140 @@ auto get_replica_value(
+            return false;
+        }
+    }
+    catch (const irods::exception& e) {
+        rodsLogError(LOG_ERROR, e.code(), e.client_display_what());
+        return false;
+    }
+    catch (const std::exception& e) {
+        rodsLogError(LOG_ERROR, SYS_UNKNOWN_ERROR, e.what());
+        return false;
+    }
+
+    return true;
+} // check_that_connected_user_is_rodsadmin
+
+auto print_pam_password_config() -> int
+{
+    if (!check_that_connected_user_is_rodsadmin(Conn)) {
+        return 1;
+    }
+
+    constexpr std::array<const char*, 4> options{
+        "password_length", "no_extend", "password_min_time", "password_max_time"};
+
+    nlohmann::json pam_password_config;
+
+    GridConfigurationInput input{};
+    std::strncpy(input.name_space, "pam_password", sizeof(input.name_space));
+
+    GridConfigurationOutput* output{};
+
+    for (const auto& opt : options) {
+        irods::at_scope_exit free_output{[&output] { std::free(output); }};
+
+        std::strncpy(input.option_name, opt, sizeof(input.option_name));
+
+        if (const auto ec = rc_get_grid_configuration_value(Conn, &input, &output); ec != 0) {
+            char* error_subname = nullptr;
+            const char* error_name = rodsErrorName(ec, &error_subname);
+            const auto msg = fmt::format(
+                "Failed to get pam_password configuration [{}]: [ec={}][msg={}]", opt, error_name, error_subname);
+
+            std::cerr << msg << "\n";
+
+            std::free(error_subname);
+
+            return 1;
+        }
+
+        pam_password_config[opt] = output->option_value;
+    }
+
+    std::cout << pam_password_config.dump(4) << '\n';
+
+    return 0;
+} // print_pam_password_config
+
+auto set_pam_password_config(const char* _option_name, const char* _option_value) -> int
+{
+    if (!check_that_connected_user_is_rodsadmin(Conn)) {
+        return 1;
+    }
+
+    if (!_option_name || !_option_value) {
+        std::cerr << "pam_password configuration option name or value is null.\n";
+        return 1;
+    }
+
+    // clang-format off
+    constexpr std::array<const char*, 4> accepted_option_names{
+        "no_extend",
+        "password_length",
+        "password_max_time",
+        "password_min_time"
+    };
+    // clang-format on
+
+    const auto option_name_is_valid = std::any_of(
+        std::begin(accepted_option_names), std::end(accepted_option_names), [_option_name](const auto* _aon) {
+            return 0 == std::strcmp(_option_name, _aon);
+        });
+
+    if (!option_name_is_valid) {
+        std::cerr << fmt::format("Invalid pam_password configuration option [{}]\n", _option_name);
+        return 1;
+    }
+
+    GridConfigurationInput input{};
+
+    if (std::strlen(_option_value) >= sizeof(input.option_value)) {
+        std::cerr << fmt::format("Invalid pam_password configuration value: Must be shorter than 2700 characters.\n");
```

Comment:
2700 hard coded here...  probably never going to change, but could be 'derived', same as the check in the line above...

Reply:
Yep, will do

Reply:
Done, I think